test/dynamo/test_decorators.py

```diff
@@ -211,6 +211,421 @@ def test_incorrect_usage_disallow_in_graph(self):
             def fn1(x):
                 return x.cos()
 
+    def test_mark_traceable_tensor_args(self):
+        @torch._dynamo.mark_traceable
+        def trace_me(x, y, z):
+            torch._dynamo.graph_break()
+            return x * y + z
+
+        def fn(x, y):
+            t0 = x + 1
+            t1 = trace_me(x, y, t0)
+            t2 = t1 + y
+            return t0 * t2
+
+        x, y = torch.randn(10), torch.randn(10)
+        opt_fn = torch.compile(fn, fullgraph=True, backend="aot_eager")
```

Comment:
Can we have a few `self.assertExpectedInline` tests? Easier to reason about when you have a correct answer in front of you.

Reply:
@zou3519 [requested this](https://github.com/pytorch/pytorch/pull/146367/files#r1941307359) and I added one to `test_flat_apply.py` as suggested. My main concern about those tests is that the captured graph is kinda an impl detail and subject to change.

Reply:
We should put all the assertExpectedInline tests into one file (test_flat_apply). The point of them is that it's easy to modify them if something does change (use the envvar)